### PR TITLE
feat: adds space-finder autocomplete combobox

### PR DESCRIPTION
**Pull request by olizilla**
Riffing on #266 

- add a space-finder combobox based on https://headlessui.com/react/combobox
- updates w3console layout to look fancy

w3console

### CODE REVIEW

**Comment by jchris** [comment]
@cmunns just want to get your eye on this animation of the new space picker UI 😀

File: packages/react/src/SpaceFinder.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          (space.name() || space.did())
+            .toLowerCase()
+            .replace(/\s+/g, '')
+            .includes(query.toLowerCase().replace(/\s+/g, ''))
+        )
+
+  return (
+    <div className={className}>
+      <Combobox value={selected} onChange={setSelected} by={(a, b) => a.sameAs(b)}>
+        <div className="relative mt-1">
```
Comment (travis) [inline packages/react/src/SpaceFinder.tsx:23]: I'm down to merge this as-is, but just a note that we'll need to translate this Tailwind styling to CSS-variable-based styling before we ship this package for real

File: packages/react/src/SpaceFinder.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import React, { Fragment, useState } from 'react'
+import { Combobox, Transition } from '@headlessui/react'
+import { CheckIcon, ChevronUpDownIcon } from '@heroicons/react/20/solid'
+
+export function SpaceFinder ({ spaces, selected, setSelected, className }) {
+  const [query, setQuery] = useState('')
+
+  console.log('space', query,  selected)
```
Comment (travis) [inline packages/react/src/SpaceFinder.tsx:8]:
```suggestion
```

**Review by travis — APPROVED** [review]

**Comment by travis** [comment]
chatted with @olizilla out of band to double check he's happy with my changes in here - good to merge!

**Pull request merged**